forensics/Copy-VmDigitalEvidence.ps1

```diff
@@ -42,152 +42,190 @@ $ErrorActionPreference = 'Stop'
+    $targetStorageContextBlob = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSAblob).Context
+    $targetStorageContextFile = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSAfile).Context
+
+    Write-Output "Start Copying Blob $SnapshotName"
+    Start-AzStorageBlobCopy -AbsoluteUri $snapSasUrl.AccessSAS -DestContainer $destSAContainer -DestContext $targetStorageContextBlob -DestBlob "$SnapshotName.vhd" -Force
 
-    Start-AzStorageFileCopy -AbsoluteUri $dsnapSasUrl.AccessSAS -DestShareName $destTempShare -DestContext $targetStorageContext -DestFilePath $dsnapshotName  -Force
-    
-    Get-AzStorageFileCopyState -Context $targetStorageContext -ShareName $destTempShare -FilePath $dsnapshotName -WaitForComplete
+    Write-Output "Start Copying Fileshare"
+    Start-AzStorageFileCopy -AbsoluteUri $snapSasUrl.AccessSAS -DestShareName $destTempShare -DestContext $targetStorageContextFile -DestFilePath $SnapshotName -Force
 
-    $ddiskpath = "$targetLinuxDir/$dsnapshotName"
-    
-    $dhashfull = Invoke-Expression -Command "sha256sum $ddiskpath"
-    $dhash = $dhashfull.split(" ")[0]
+    Write-Output "Waiting Fileshare Copy End"
+    Get-AzStorageFileCopyState -Context $targetStorageContextFile -ShareName $destTempShare -FilePath $SnapshotName -WaitForComplete
 
-    Write-Output "Computed SHA-256: $dhash"
+    # Linux hash version if you use a Linux Hybrid Runbook Worker
+    $diskpath = "$targetLinuxDir/$snapshotName"    
+    Write-Output "Start Calculating HASH for $diskpath"
+    $hashfull = Invoke-Expression -Command "sha256sum $diskpath" 
+    $hash = $hashfull.split(" ")[0]
+    Write-Output "Computed SHA-256: $hash"
 
-    Start-AzStorageBlobCopy -AbsoluteUri $dsnapSasUrl.AccessSAS -DestContainer $destSAContainer -DestContext $targetStorageContext -DestBlob $dsnapshotName -Force
-    
-    $BEKurl = $ddisk.EncryptionSettingsCollection.EncryptionSettings.DiskEncryptionKey.SecretUrl
+    #################### Copy the OS BEK to the SOC Key Vault  ###################################
+    $BEKurl = $disk.EncryptionSettingsCollection.EncryptionSettings.DiskEncryptionKey.SecretUrl
     Write-Output "#################################"
-    Write-Output "Disk Encryption Secret URL: $BEKurl"
+    Write-Output "OS Disk Encryption Secret URL: $BEKurl"
     Write-Output "#################################"
     if ($BEKurl) {
         $sourcekv = $BEKurl.Split("/")
         $BEK = Get-AzKeyVaultSecret -VaultName  $sourcekv[2].split(".")[0] -Name $sourcekv[4] -Version $sourcekv[5]
         Write-Output "Key value: $BEK"
-        Write-Output "Secret name: $dsnapshotName"
-        Set-AzKeyVaultSecret -VaultName $destKV -Name $dsnapshotName -SecretValue $BEK.SecretValue -ContentType "BEK" -Tag $BEK.Tags
+        $BEK.Tags.Hash = $hash
+        Get-AzSubscription -SubscriptionId $destSubId | Set-AzContext
+        $secretName = $snapshotName.Replace("_","-")

```

Comment:
Idea: should we just do this up where the snapshot name is generated so it's consistent across all usages, not just `$secretName`? [Same Q for data disk below]

Reply:
done